### DISCUSSION TITLE
Time for a more robust DB Engine?

### DISCUSSION THREAD

**Discussion by RobbieE-ZA** [Chia-Network/chia-blockchain#18953] [Ideas]
I've been running a Chia farm for a couple of years now, and the number of times I've experienced the "sqlite3.DatabaseError: database disk image is malformed" error is depressingly.

I live in an area where the power is not always so reliable and can be off for hours at a time, so, even stored power eventually runs out if I'm not at home to turn the generator on.

Replacing the database usually happens by two routes:
1. Downloading the latest image from the official site, or
2. Deleting the blockchain and resynchronising from scratch. I actually did this once and it took more than 5 months to get up to date, not because of my internet connection (it's a reliable 100Mbps line), but rather the super-throttling that peers apply.

I've never known any other database engine this fragile, and with the blockchain now almost 180GB, wouldn't it be prudent to move to a more robust engine?

**Comment by BrandtH22** [comment]
Hey @RobbieE-ZA , we are looking into some ways to expedite the process of loading a clean db for when the new plot format launches. Until then some good things to look into are:
- what OS you are using, windows tends to lead to more corrupt dbs than linux or mac (personally I use linux and have never had a corrupt db even with multiple power outages)
- power backup/redundancy, it is generally a good idea to have a backup power source in place for unexpected power outages. That way you can have more time to recognize the power outage has occurred and shut down the farm smoothly
- keeping a backup, when you download the db snapshot it might be worth storing a backup of it in case you need to reload from it again in the future. Generally this should not be needed but if you are having to load from the db frequently this would save time in the long run
